common/flatpak-oci-registry.c

```diff
@@ -68,7 +68,7 @@ struct FlatpakOciRegistry
 
   /* Remote repos */
   SoupSession *soup_session;
-  SoupURI     *base_uri;
+  GUri        *base_uri;
```

Comment:
Unfortunately, this is going to break the build when using older GLib versions.

Flatpak aims to be compatible with old versions of GLib, so that backports for old versions of LTS distributions like RHEL/CentOS, Debian and Ubuntu are possible.

Reply:
Right, that's why I bumped the glib version requirement. Having conditional compile also on this thing would make the code close to unreadable, I'm afraid.

Reply:
2.66 is in Debian 11, and will be in Ubuntu 22.04 LTS, but is not in Ubuntu 20.04; so we cannot have a GLib 2.66 dependency until/unless we are ready to lose the ability to backport to Debian 10 and Ubuntu 20.04.

Meanwhile in Red Hat-world, CentOS 8 appears to be on GLib 2.56, so no backports there either.

Reply:
Yes, I understand your point, that's no problem (though not every change touches the libsoup code and even that could be backported, only with a non-trivial effort). I'm asking for available options. Having GUri/SoupURI conditional code will make it hard to read and easy to break (as some parts could be unintentionally missed in the "disabled" code).